src/tbb/arena_slot.cpp

```diff
@@ -96,52 +102,40 @@ d1::task* arena_slot::get_task(execution_data_ext& ed, isolation_type isolation)
                 release_task_pool();
             }
         }
-        result = get_task_impl( T, ed, tasks_omitted, isolation );
+        result = get_task_impl( T, ed, tasks_skipped, isolation );
         if ( result ) {
-            poison_pointer( task_pool_ptr[T] );
+            // If some tasks were skipped, we need to make a hole in position T.
+            if ( tasks_skipped ) {
+                task_pool_ptr[T] = nullptr;
+            } else {
+                poison_pointer( task_pool_ptr[T] );
+            }
             break;
-        } else if ( !tasks_omitted ) {
+        } else if ( !tasks_skipped ) {
             poison_pointer( task_pool_ptr[T] );
             __TBB_ASSERT( T0 == T+1, nullptr );
             T0 = T;
         }
     } while ( !result && !task_pool_empty );
 
-    if ( tasks_omitted ) {
+    if ( tasks_skipped ) {
+        __TBB_ASSERT( is_task_pool_published(), nullptr );
         if ( task_pool_empty ) {
-            // All tasks have been checked. The task pool should be  in reset state.
-            // We just restore the bounds for the available tasks.
-            // TODO: Does it have sense to move them to the beginning of the task pool?
-            __TBB_ASSERT( is_quiescent_local_task_pool_reset(), nullptr );
+            // All tasks have been checked.
             if ( result ) {
                 // If we have a task, it should be at H0 position.
                 __TBB_ASSERT( H0 == T, nullptr );
                 ++H0;
             }
             __TBB_ASSERT( H0 <= T0, nullptr );
```

Comment:
Seemingly, there is nothing useful done in the `task_pool_empty` branch anymore, as it only has assertions and invariant preparations for these. Perhaps It can be removed completely.